n3fit/src/n3fit/layers/observable.py

```diff
@@ -15,6 +17,44 @@ def is_unique(list_of_arrays):
     return True
 
 
+def compute_pdf_boundary(pdf, q0_value, xgrid, n_std, n_replicas):
+    """
+    Computes the Unpolarized PDF set to be used as a Boundary Condition and
+    convert it into a Tensor object that can be understood by the convolution.
+
+    Parameters
+    ----------
+    pdf: validphys.core.PDF
+        a validphys instance of the Unpolarized PDF set
+    q0_value: float
+        starting scale of the theory as defined in the FK tables
+    xgrid: np.ndarray
+        a grid containing the x-values to be given as input to the PDF
+    n_std: int
+        integer representing the shift to the CV w.r.t. the standard
+        deviation
+    n_replicas: int
+        number of replicas fitted simultaneously
+
+    Returns
+    -------
+    tf.tensor:
+        a tensor object that has the same shape of the output of the NN
+    """
+    xpdf_obj = xplotting_grid(pdf, q0_value, xgrid, basis="FK_BASIS")
+    # Transpose: take the shape from (n_fl, n_x) -> (n_x, n_fl)
+    xpdf_cvs = xpdf_obj.grid_values.central_value().T
+    xpdf_std = xpdf_obj.grid_values.std_error().T
+
+    # Computes the shifted Central Value as given by `n_std`
+    xpdf_bound = xpdf_cvs + n_std * xpdf_std
+
+    # Expand dimensions for multi-replicas and convert into tensor
+    mult_resx = np.repeat([xpdf_bound], n_replicas, axis=0)
+    add_batch_resx = np.expand_dims(mult_resx, axis=0)
+    return op.numpy_to_tensor(add_batch_resx)
```

Comment:
Please add a test for this function where you call it with a PDF and check on the other side that the result has the right shape with the values in the right positions.